### ISSUE TITLE
Build issues on macOS

### ISSUE THREAD

**Issue by llighterr** [WandererXII/shoginet#1]
Hi!
I'm so excited to see the new **shoginet** feature, and how fast **lishogi** evolves.😊

Now about the build. I was following the [how to setup](https://github.com/WandererXII/shoginet#how-to-setup) instructions,
and found out the build script is not supporting Mac well.

It requires a few changes to be fully compatible:
1) [this check](https://github.com/WandererXII/shoginet/blob/main/build-yaneuraou.sh#L19) should be updated with Mac-specific logic
2) It requires `gcc` by default. It took me some time to understand why the build is failing. Maybe it makes sense to point that in README.
3) On [this line ](https://github.com/WandererXII/shoginet/blob/main/shoginet.py#L1018) it expects a bit different naming convention for Mac. Probably I messed up the `fishnet.ini` config file. Forcing the `YaneuraOu-by-gcc` helped here as a quick fix.

But after that, it worked like a charm.👌

To summarize, all the issues are not hard to fix.
I think I could even create PR with the changes, just wanted to discuss it beforehand.👍🏻

**Comment by WandererXII** [comment]
Hey, thanks a lot for taking interest and helping with the project. If you make a PR, it would be really helpful, I could try fixing some of the things you mentioned, but I don't have the means to test in on macOS.

**Comment by llighterr** [comment]
Fixed in #2
